projects/baby_driver/scripts/adc_read.py

```diff
@@ -0,0 +1,58 @@
+"""Python implementation of the adc_read function for Babydriver."""
+
+import can_util
+from gpio_port import GpioPort
+from message_defs import BabydriverMessageId
+
+NUM_PINS_PER_PORT = 16
+
+def adc_read(port, pin, raw):
+    """
+    Returns a raw or converted ADC reading on a specific port and pin from the firmware project.
+
+    The port can be entered as either a string or int value (e.g. 'A' or 0).
+    The pin is an int value, and raw is a bool value.
+    """
+
+    # If port is entered as a str, convert to int
+    if isinstance(port, str):
+        port = getattr(GpioPort, port.capitalize())
+
+    # Check the ranges of input args
+    if port < 0 or port >= GpioPort.NUM_GPIO_PORTS:
+        raise ValueError("ERROR: invalid GPIO port")
+    if pin < 0 or pin >= NUM_PINS_PER_PORT:
+        raise ValueError("ERROR: invalid GPIO pin number")
+    if raw not in (True, False):
+        raise ValueError("ERROR: raw must be a bool value")
+
+    # Send CAN message
+    data = [
+        (BabydriverMessageId.ADC_READ_COMMAND, 1),
```

Comment:
If you specify the `babydriver_id` keyword parameter like `babydriver_id=BabydriverMessageId.ADC_READ_COMMAND` in the `send_message` call, you won't have to specify this here; the babydriver ID is added onto the beginning of the packed data for you.